src/bp/ui-studio/src/web/views/OneFlow/diagram/VariablesEditor/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { Button, FormGroup, InputGroup } from '@blueprintjs/core'
+import axios from 'axios'
+import { FlowVariable } from 'botpress/sdk'
+import { Dialog } from 'botpress/shared'
+import { FlowView } from 'common/typings'
+import React, { FC, Fragment, useEffect, useState } from 'react'
+import { connect } from 'react-redux'
+import { fetchFlows } from '~/actions'
+import { getCurrentFlow, RootReducer } from '~/reducers'
+
+import VariableModal from './Modal'
+
+interface OwnProps {}
+type StateProps = ReturnType<typeof mapStateToProps>
+type DispatchProps = typeof mapDispatchToProps
+
+type Props = StateProps & DispatchProps & OwnProps
+
+const VariablesEditor: FC<Props> = props => {
+  const [editingVariable, setEditingVariable] = useState<FlowVariable>(undefined)
+
+  const affectUpdateFlow = async (flow: FlowView) => {
+    await axios.post(`${window.BOT_API_PATH}/flow/${flow.location}`, { flow })
+    props.fetchFlows()
+  }
+
+  const computeCreateVariable = (flow: FlowView, newVariable: FlowVariable) => {
+    if (!flow.variables) {
+      flow.variables = [newVariable]
+    } else {
+      flow.variables.push(newVariable)
+    }
+  }
+
+  const computeDeleteVariable = (flow: FlowView, variableName: string) => {
+    flow.variables = flow.variables.filter(v => v.name != variableName)
+  }
+
+  const computeEditVariable = (flow: FlowView, variableName: string, newValues: FlowVariable) => {
+    const index = flow.variables.findIndex(v => v.name == variableName)
+    flow.variables[index] = newValues
+  }
+
+  const handleCreateClick = async () => {
+    const newVariable: FlowVariable = {
+      type: 'string',
+      name: `New Variable ${props.currentFlow.variables?.length ?? 0}`,
+      description: 'A new variable'
```

Comment:
There's no way to change the description, also i'd keep it empty by default. Also, the name of a variable can't have spaces in it, must match [a-zA-Z0-9-_]